Source/WebCore/platform/mediastream/mac/ScreenCaptureKitCaptureSource.mm

```diff
@@ -651,6 +653,88 @@ - (void)outputVideoEffectDidStopForStream:(SCStream *)stream
     return CaptureDevice(String::number(windowID.value()), CaptureDevice::DeviceType::Window, emptyString(), emptyString(), true);
 }
 
+RetainPtr<CGImageRef> ScreenCaptureKitCaptureSource::captureWindowSnapshot(CGWindowID windowID, CGRect rect, OptionSet<ScreenCaptureKitCaptureSource::SnapshotOptions> options)
+{
+#if HAVE(SCREEN_SHOT_MANAGER)
+    RetainPtr<CGImageRef> captureImage = nullptr;
+    RetainPtr<NSError> captureError = nullptr;
+
+    dispatch_semaphore_t semaphore = dispatch_semaphore_create(0);
+
+    auto completionHandler = makeBlockPtr([&] (SCShareableContent* shareableContent, NSError* error) mutable {
+        if (error || !shareableContent) {
+            RELEASE_LOG_ERROR(WebRTC, "ScreenCaptureKitCaptureSource::captureWindowSnapshot: +[SCShareableContent getCurrentProcessShareableContentWithCompletionHandler:] failed with error %s", error.localizedDescription.UTF8String);
+            dispatch_semaphore_signal(semaphore);
+            return;
+        }
+
+        SCWindow* targetWindow = nullptr;
```

Comment:
`RetainPtr<SCWindow>` ?

Reply:
`shareableContent.windows` is a an NSDictionary, which already has a retain on all of the windows and we won't use it outside of the lambda.